Code/GraphMol/new_canon.cpp

```diff
@@ -594,6 +603,9 @@ void updateAtomNeighborNumSwaps(
 
 void rankMolAtoms(const ROMol &mol, std::vector<unsigned int> &res,
                   bool breakTies, bool includeChirality, bool includeIsotopes) {
+  if (!mol.getRingInfo()->isInitialized()) {
+    MolOps::fastFindRings(mol);
```

Comment:
This looks to be a side-effect now and uses fastFingRings in place of getSymmSSSR.  Is this a copy of a mol?  Should this be "uninitialized" at the end?

Reply:
As an aside, part of me wants to make getRingInfo() make the ring info by default (i.e. use a lock and a mutable data structure if necessary).  it will still be "const" since ROMols are const.

Reply:
I agree that it's worth thinking about doing something more sensible with the ring finding code and RingInfo.
And, yes, this probably should be "unintializing" the RI data structure.